### PR TITLE
fix(metrics): compact ruleStats

### PR DESCRIPTION
**Pull request by amchiclet**
Partially helps: PA-3321.

Follow up from https://github.com/semgrep/semgrep/pull/9477.

Tested with `make test` locally under cli to make sure there were no regressions. Some other tests were failing, but those tests were also failing on a fresh clone.

Also tested by checking semgrep scan results in metabase.
* For osemgrep, I ran with experiment and profile flags. 
* For pysemgrep, I hardcoded some examples to trigger the code that populates ruleStat (because I was unable to [make this condition be true](https://github.com/semgrep/semgrep/blob/2b36cd5b04b7b06741ab9cbed8475900ed226d03/cli/src/semgrep/metrics.py#L235-L252))

For both cases, the rules with bytes scanned and matching time greater than 0 showed up in the metrics when querying metabase.

### CODE REVIEW

**Comment by aryx** [comment]
What tests are failing locally on a fresh clone?
We need to fix those!
I'm on Linux and all tests are passing locally. I remember jonas actually said some tests were failing on macOS locally.

File: cli/src/semgrep/metrics.py
Comment (aryx) [inline cli/src/semgrep/metrics.py:254]: maybe would be good to have some tests for that (the time will differ between runs, but we have some code
in pytest to mask times.
That way we can also mark this tests as 'osemfail' so we know we need to port it.

**Review by aryx — APPROVED** [review]

File: cli/src/semgrep/metrics.py
Comment (amchiclet) [reply cli/src/semgrep/metrics.py:254]: Good idea. I'll try to add that change before merging.

Comment (amchiclet) [reply cli/src/semgrep/metrics.py:254]: But first, I probably need to figure out how to trigger this code path in pysemgrep.

Comment (amchiclet) [reply cli/src/semgrep/metrics.py:254]: Done. I was able to piggyback on one of the tests that test the metrics payload. Waiting for CI to merge.

**Pull request merged**
